manual/src/tutorials/gadtexamples.etex

```diff
@@ -0,0 +1,305 @@
+\chapter{Generalized algebraic datatypes} \label{c:gadts-tutorial}
+%HEVEA\cutname{gadts-tutorial.html}
+
+\begin{caml_example*}{verbatim}
+type _ term =
+  | Int : int -> int term
+  | Add : (int -> int -> int) term
+  | App : ('b -> 'a) term * 'b term -> 'a term
+\end{caml_example*}
```

Comment:
What an abrupt start...  I know this is taken verbatim from the current "language extensions" chapter, but for a tutorial it's really too abrupt.  

In my lectures, I like to use the following two examples.  First for existential quantification of variables:
```
  type printable = 
    | Printable: 'a * ('a -> string) -> printable
```
Second for "type parameters that change depending on the value constructor":
```
  type 'a compact_array =
  | Array: 'a array -> 'a compact_array  (* default case *)
  | Bytes: bytes    -> char compact_array (* optimized case *)
```